kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -392,4 +372,57 @@ private Map<String, DomainMetadata> loadDomainMetadataMap(Engine engine) {
       throw new UncheckedIOException("Could not close iterator", ex);
     }
   }
+
+  // Calculates the latest snapshot hint before current snapshot version, returns the CRCInfo if
```

Comment:
Some thoughts:

1. Please use `/** */` for multi-line methoddocs, no `//`

2. `Calculates the latest snapshot hint before current snapshot version` --> below, you also return it if `snapshotHint.version == snapshotVersion`. So I don't think "before" is correct, right?

Reply:
Change it to "before or at the current snapshot version". Thanks for catching this boundary case.